flocker/node/agents/ebs.py

```diff
@@ -107,6 +115,68 @@ def _wait_for_volume(expected_volume,
             )
 
 
+def _check_blockdevice_size(device, size):
+    """
+    Helper function to check if the size of block device with given
+    suffix matches the input size.
+
+    :param unicode device: Name of the block device to check for size.
+    :param int size: Size, in SI metric bytes, of device we are interested in.
+
+    :returns True if a block device with given name has given size.
+        False otherwise.
+    """
+    device_name = b"/dev/" + device
+    command = [b"lsblk", b"-nb", b"-o", b"SIZE", device_name]
+    device_size = int(check_output(command).strip().decode("ascii"))
+
+    # OS reports device size in powers of 2, where has EBS
+    # talks powers of 10. Hence, convert before comparing.
+    if size/(1000**3) == device_size/(1024**3):
+        return True
+    else:
+        return False
+
+
+def _wait_for_new_device(base, size, time_limit=60):
+    """
+    Helper function to wait for up to 60s for new
+    EBS block device (`/dev/sd*` or `/dev/xvd*`) to
+    manifest in the OS.
+
+    :param list base: List of baseline block devices
+        that existed before execution of operation that expects
+        to create a new block device.
+
+    :returns unicode device: Name of the new block device
```

Comment:
`:returns unicode device:` won't be valid markup.  I'm still not sure whether `:returns unicode:` is or not (I think possibly you have to write `:returns: ...` and separately `:rtype: ...` though I've started writing `:returns unicode:` with the optimistic viewpoint that we'll just fix the tool chain to support that form, since it's easier to write).